src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function sortGoods(goods, query, reversed) {
+  let listOfGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (query) {
+    case 'alphabet':
+      listOfGoods = [...goods].sort((good1, good2) => {
+        return good1.localeCompare(good2);
+      });
+      break;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    case 'length':
+      listOfGoods = [...goods].sort(
+        (good1, good2) => good1.length - good2.length,
+      );
+      break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    default:
+      break;
+  }
+
+  if (reversed) {
+    return listOfGoods.reverse();
+  }
+
+  return listOfGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const showResetButton = sortField || isReversed;
+  const goods = [...goodsFromServer];
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== 'alphabet',
+          })}
+          onClick={() => setSortField('alphabet')}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== 'length',
+          })}
+          onClick={() => setSortField('length')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {showResetButton && (
+          <button
+            type="button"
+            className={cn('button', 'is-danger', {
+              'is-light': sortField,
```

Comment:
The condition `'is-light': sortField` should be `'is-light': !sortField` to correctly apply the 'is-light' class when no sort field is selected, aligning with the logic used in other buttons.